### PR TITLE
Feat : Added WIshlist Management

### PR DESCRIPTION
**Pull request by shlok30**
Hi Team,

I have added the following features to my app

1)Allow user to add item to wishlist only if the user is signed in
2)Add product to wishlist from the product listing page
3)Update quantity of items in the wishlist in the nav bar icon
4)Once you click on 'Add to Wishlist' the button text changes to 'Go to Wishlist'
5)In the cart page you can add an item to wishlist or remove it from wishlist if its already present in the wishlist.
6)Render all the items present in the wishlist on the wishlist page 

Live Link : https://water-comm-react-gj3dyz24u-shlok30.vercel.app/products

Demo Login Credentials
Username : test@gmail.com
Password : test 

Please let me know what you think

Thanks

### CODE REVIEW

File: my-app/src/components/cart-card.js [hunk 1]
```diff
@@ -20,7 +31,7 @@ const CartCard = ({qty,price,author,id,title}) => {
                         <button style = {{width:'1rem'}} onClick = {() => updateQuantity(id,state.encodedToken,'increment',dispatch,qty)}>+</button>
                         <button style = {{width:'1rem'}} disabled = {qty === 1} onClick = {() => updateQuantity(id,state.encodedToken,'decrement',dispatch,qty)}>-</button>
                         <button className = "btn btn-error m2-top full-width" onClick = {() => deleteProduct(id,state.encodedToken,dispatch)}>Remove from Cart</button>
```
Comment (shrushti2000) [inline my-app/src/components/cart-card.js:33]: Good job on disabling the decrement button after the count goes below 1. I will add this to my ecom app  as well.

File: my-app/src/components/nav.js [hunk 2]
```diff
@@ -26,7 +31,10 @@ const Nav = () => {
             </div>
             <div className = "flex gap-xl flex-center grow-1" >
                 <Link to = '/login'><button className="btn btn-secondary">Login</button></Link>
-                <i className="material-icons cursor-pointer" >favorite_border</i>
+                <div class = "icon badge" onClick = {handleClick} style = {{cursor : "pointer"}}>
```
Comment (shrushti2000) [inline my-app/src/components/nav.js:34]: Can avoid inline style here.

File: my-app/src/context/functions/addToWishlist.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+const addToWishlist = (product,encodedToken,dispatch) => {
+    
+    axios
+    .post("/api/user/wishlist",{"product" : product},{headers : { 'authorization' : encodedToken}})
+    .then((res) => {
+        console.log("successfully added to Wislist",res.data)
+        dispatch({type : "WISHLIST" , payload : res.data.wishlist})
+    })
+    .catch((err) => console.log(err))
```
Comment (tushargoyal812) [inline my-app/src/context/functions/addToWishlist.js:12]: suggestion: use try catch here its a good practice

File: my-app/src/context/functions/addToCart.js [hunk 4]
```diff
@@ -1,13 +1,12 @@
 import React from 'react'
 import axios from "axios"
 
-const addToCart = ({id,title,author,price},encodedToken,dispatch) => {
-    //console.log("From add to cart",{'_id' : id , 'title' : title , 'author' : author , 'price' : price})
+const addToCart = ({id,title,author,price,category,rating},encodedToken,dispatch) => {
+   
     axios
-     .post("/api/user/cart",{"product" : { "_id" : id , "title" : title , "author" : author , "price" : price}},{headers :{ 'authorization' : encodedToken}})
+     .post("/api/user/cart",{"product" : { "_id" : id , "title" : title , "author" : author , "price" : price , "category" : category , "rating" : rating}},{headers :{ 'authorization' : encodedToken}})
```
Comment (shrushti2000) [inline my-app/src/context/functions/addToCart.js]: In places where the key and values are same ,no need to do "title":title . 
You can simply do {"_id" : id ,  title ,  author ,  price , category , rating}

File: my-app/src/context/functions/removeFromWishlist.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import axios from 'axios'
+
+const removeFromWishlist = (id,encodedToken,dispatch) => {
+    console.log("Id from removewishlist",id)
```
Comment (tushargoyal812) [inline my-app/src/context/functions/removeFromWishlist.js:5]: remove console

**Review by tushargoyal812 — COMMENTED** [review]
overall code is good and to the point.
liked how you structured files.
variable names are also good.

File: my-app/src/context/user-context.js [hunk 6]
```diff
@@ -3,14 +3,16 @@ import userReducer from "./reducers/userReducer";
...
 import updateQuantity from "./functions/updateQuantityInCart";
 import deleteProduct from "./functions/deleteFromCart";
+import addToWishlist from "./functions/addToWishlist";
+import removeFromWishlist from "./functions/removeFromWishlist";
 
 const UserContext = createContext()
 
 const UserContextProvider = ({children}) => {
     const [state,dispatch] = useReducer(userReducer,{encodedToken : "" , cart : [] , wishlist : [],showLoginAlert : false})
-    console.log("user context state",state)
```
Comment (shrushti2000) [inline my-app/src/context/user-context.js:11]: I liked how you have used the showLoginAlert state to handle unauthorized users.

**Review by shrushti2000 — APPROVED** [review]
I liked few state declarations that you have made. All the features are working properly on live preview. Good work.

**Pull request merged**